### PR TITLE
Issue 2800 - Add logic to check for unsupported sdo key names

### PR DESCRIPTION
**Pull request by Polber**
Signed-off-by: polber <jeff@thekinards.com>

### CODE REVIEW

File: cli/sdo/key.go [hunk 1]
```diff
@@ -287,3 +297,25 @@ func checkEmptyKeyFields(key KeyFile) error {
...
+        if (r < 'a' || r > 'z') && (r < '0' || r > '9') && r != '-' {
+            return errors.New("Key name can only contain lowercase characters, numbers, and hyphens")
+        }
+    }
+	return nil
+}
+
+func checkKeyCountry(s string) error {
+	err := "Country name must be a 2 Letter Country Code"
+	if len(s) > 2 {
```
Comment (bmpotter) [inline cli/sdo/key.go:312]: My understanding is that the country code must be exactly 2 chars, so just check `len(s) == 2`

File: test/gov/hzn_sdo.sh [hunk 2]
```diff
@@ -67,6 +67,30 @@ cat <<'EOF' > /tmp/sdo_key2.json
 }
 EOF
 
+cat <<'EOF' > /tmp/sdo_key_unsupported.json
```
Comment (bmpotter) [inline test/gov/hzn_sdo.sh:70]: Why is this metadata unsupported and why don't i see it used anywhere?

**Review by bmpotter — CHANGES_REQUESTED** [review]

File: test/gov/hzn_sdo.sh (hunk 2 above)
Comment (Polber) [reply test/gov/hzn_sdo.sh:70]: Because I used the wrong name in the test. Thanks for the catch.

File: cli/sdo/key.go (hunk 1 above)
Comment (Polber) [reply cli/sdo/key.go:312]: It should be `len(s) != 2`. I'm not sure when or why I changed it to `>` but that is definitely incorrect.

**Review by bmpotter — DISMISSED** [review]

**Review by bmpotter — APPROVED** [review]

**Pull request merged**